src/components/app.js

```diff
@@ -12,7 +12,7 @@ import SearchBar from './searchbar'
 const App = React.createClass({
   componentDidMount() {
     const query = { ...this.props.query }
-    const hash = decodeURIComponent(window.location.hash)
+    const hash = decodeURIComponent(window.location.hash.replace(/&/g, ';'))
```

Comment:
One thing that I like about this approach is that it maintains backwards compatibility with all the old blockbuilder search urls, where the entire url hash was URI encoded.  This works since the only transformation we do is to replace the `&` character,  which should never appear in those urls.

Reply:
Great idea for backwards compatibility!